src/transformStateWithClones.js

```diff
@@ -1,13 +1,37 @@
 'use strict';
-
 /**
  * @param {Object} state
  * @param {Object[]} actions
  *
  * @return {Object[]}
  */
-function transformStateWithClones(state, actions) {
-  // write code here
+
+function transformStateWithClones(initialState, actions) {
+  const stateHistory = [];
+  let state = { ...initialState };
```

Comment:
[NAMING]: Instead of naming your object copy as 'state', it's clearer to name it 'stateCopy' to indicate that it is a copy of the state.